src/lualib/ArrayFlat.ts

```diff
@@ -0,0 +1,12 @@
+function __TS__ArrayFlat(this: void, array: any[], depth = 1): any[] {
```

Comment:
Does this break if you type it `function __TS__ArrayFlat<T>(this: void, array: Array<T[]>, depth = 1): T[] {`? That seems more correct to me

Reply:
It's not correct because of the depth. Without some complex types it's impossible to correctly type it (TS now supports only depth up to 4, and falls back to `any[]` - https://github.com/Microsoft/TypeScript/blob/f7c4fefeb62416c311077a699cc15beb211c25c9/src/lib/es2019.array.d.ts#L112)